android/src/main/java/com/reactnativecommunity/webview/jsi/WebViewJsiInterface.java

```diff
@@ -5,5 +5,13 @@
     System.loadLibrary("reactnativecommunity-webview");
   }
 
-  public native boolean onShouldStartLoadWithRequest(long javaScriptContextHolder, String key, String url, boolean loading, String title, boolean canGoBack, boolean canGoForward);
+  public native boolean onShouldStartLoadWithRequest(
+    long javaScriptContextHolder,
+    String key,
+    String url,
+    boolean loading,
+    String title,
+    boolean canGoBack,
+    boolean canGoForward)
+    throws JsiJSError, JsiException;
```

Comment:
Maybe I'm a bit rusty on Java, but do you have to say it can throw subclasses? Or is it best practice? `JsiJSError` is a `JsiException`.

Reply:
I guess we can just declare that it throws `JsiJSError`. If we put only `JsiException` then we couldn't catch `JsiJSError` but if we only put `JsiJSError` we can catch both because `JsiJSError` extends `JsiException`

Reply:
Both exceptions could be thrown, so I put both as documentation. You could just put JsiException but then it is not immediately obvious that 2 subtly different but related types of exceptions can be thrown. The client of this class could want to handle JSErrors and JSIExceptions differently. I put both here as a form of documentation.